File: app/src/main/java/com/example/theloop/models/WeatherResponse.java
package com.example.theloop.models;

import com.google.gson.annotations.SerializedName;

public class WeatherResponse {

    @SerializedName("latitude")
    private double latitude;

    @SerializedName("longitude")
    private double longitude;

    @SerializedName("current")
    private CurrentWeather current;

    @SerializedName("daily")
    private DailyWeather daily;

    // Getters
    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public CurrentWeather getCurrent() {
        return current;
    }

    public DailyWeather getDaily() {
        return daily;
    }
}

File: app/src/main/java/com/example/theloop/models/WeatherResponse.kt
package com.example.theloop.models

import com.google.gson.annotations.SerializedName

data class WeatherResponse(
    @SerializedName("latitude") val latitude: Double,
    @SerializedName("longitude") val longitude: Double,
    @SerializedName("current") val current: CurrentWeather,
    @SerializedName("daily") val daily: DailyWeather?
)
